File: LeetCode/cpp/1249_Minimum_Remove_to_Make_Valid_Parentheses.cpp
class Solution {
public:
	string minRemoveToMakeValid(string s) {
		string answer;
		int check = 0;

		for (int i = 0; i < s.length(); i++) {
			if (s[i] != '(' && s[i] != ')') {
				answer += s[i];
				continue;
			}

			if (s[i] == '(') {
				check++;
				answer += s[i];
				continue;
			}

			if (check > 0) { //s[i] == ')'
				answer += s[i];
				check--;
			}
		}

		int idx = answer.length() - 1;

		while (check > 0) {
			if (idx < 0) {
				break;
			}

			if (answer[idx] == '(') {
				answer.erase(answer.begin() + idx);
				check--;
			}

			idx--;
		}

		return answer;
	}
};

File: LeetCode/java/1249_Minimum_Remove_to_Make_Valid_Parentheses.java
class Solution {
    public String minRemoveToMakeValid(String s) {
        StringBuilder answer = new StringBuilder();
        int check = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != '(' && s.charAt(i) != ')') {
                answer.append(s.charAt(i));
                continue;
            }

            if (s.charAt(i) == '(') {
                check++;
                answer.append(s.charAt(i));
                continue;
            }

            if (check > 0) { //s.charAt(i) == ')''
                answer.append(s.charAt(i));
                check--;
            }
        }

        int idx = answer.length() - 1;

        while (check > 0) {
            if (idx < 0) {
                break;
            }

            if (answer.charAt(idx) == '(') {
                answer.deleteCharAt(idx);
                check--;
            }

            idx--;
        }

        return answer.toString();
    }
}